### PR TITLE
Add log2_ to torch frontend api with test

### LINKED ISSUE #21567: log2_

### PR DESCRIPTION
**Pull request by Sjsingh101**
closes #21567

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -2098,6 +2098,42 @@ def test_torch_log2(
     )
 
 
+# log2_
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="log2_",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2107]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -367,6 +367,11 @@ def log_(self):
     def log2(self):
         return torch_frontend.log2(self)
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16", "bfloat16")}, "torch")
+    def log2_(self):
+        self.ivy_array = self.log2().ivy_array
+        return self
```
Comment (hello-fri-end) [inline ivy/functional/frontends/torch/tensor.py:373]: Not sure I understand what you are trying to do here. Can't you just return `torch_frontend.log2(self)`?

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @Sjsingh101! I am not able to find the  `log2_` function in pytorch's documentation. Would you please drop a link for the same? Sorry for the inconvenience. Please request another review when you have resolved the comments and provided the link. Thanks :)

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 1 above)
Comment (Sjsingh101) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:2107]: @hello-fri-end Link to the documentation: https://pytorch.org/docs/stable/generated/torch.Tensor.log2_.html#torch.Tensor.log2_

File: ivy/functional/frontends/torch/tensor.py (hunk 2 above)
Comment (Sjsingh101) [reply ivy/functional/frontends/torch/tensor.py:373]: In the `log2(self)` function just above it we are returning `torch_frontend.log2(self)`. In this function, We are calling that above function and updating the array itself.

**Pull request closed without merging**